install/kubernetes/aws/eks-cluster.yaml.tpl

```diff
@@ -0,0 +1,102 @@
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+metadata:
+  name: ${CLUSTER_NAME}
+  region: ${REGION}
+  version: "1.27"
+  tags:
+    createdBy: eksctl
+    environment: dev
+    karpenter.sh/discovery: ${CLUSTER_NAME}
+
+karpenter:
```

Comment:
Prefer to set minimal number of options:

```yaml
karpenter:
  version: 'v0.29.0'
```

Reply:
I don't have confidence karpenter would work without some of these.

1. `createServiceAccount` - without this I don't think we have a karpenter role. We'd need to create our own by some other process and inject it to karpenter configuration. I'll take this line over that effort.
2. `withSpotInterruptionQueue` - we could probably get rid of this but why? From the schema doc:
> if true, adds all required policies and rules for supporting Spot Interruption Queue on Karpenter deployments
Isn't that something we and customers would likely want? This is effectively the replacement component for the node-termination-handler [docs](https://aws.amazon.com/blogs/containers/using-amazon-ec2-spot-instances-with-karpenter/).
3. `defaultInstanceProfile` - has to be the same across both the provisioner and this manifest. I think we need it or we're guessing or relying on convention of those tools.
4. `version` - :shrug: we might be able to get away without? I think version guarantees are a good thing to ensure our users get something that is known to work.